### PR TITLE
feat: New Core Lib.

### PR DESCRIPTION
**Pull request by bloiseleo**
All new commands and configuration will be loaded automatically on Initialization.

### CODE REVIEW

File: src/main/java/tech/chorume/bot/core/DiscordBotConfigurationDefault.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                .filename(".env")
+                .load();
+    }
+    @Override
+    public String getDiscordToken() {
+        var token = dotenv.get("bot.token");
+        return token;
+    }
+    /**
+     * Define os gateways padrões do Bot Discord. Por padrão, ele usa os seguintes Gateways:
```
Comment (brunofunnie) [inline src/main/java/tech/chorume/bot/core/DiscordBotConfigurationDefault.java:29]: Wrong language, should use standard american english

File: src/main/java/tech/chorume/bot/core/DiscordBotConfigurationDefault.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    @Override
+    public Collection<GatewayIntent> getGatewayIntents() {
+        //https://docs.jda.wiki/net/dv8tion/jda/api/requests/GatewayIntent.html
+        return Set.of(GatewayIntent.GUILD_MESSAGES, GatewayIntent.DIRECT_MESSAGES);
+    }
+
+    /**
+     * Define os listeners padrões do Bot Discord. Por padrão, ele registra o dispatcher default do bot, que tem um mapa
+     * que lida com todos os comandos previamente determinados.
```
Comment (brunofunnie) [inline src/main/java/tech/chorume/bot/core/DiscordBotConfigurationDefault.java:42]: Wrong language, should use standard american english

File: src/main/java/tech/chorume/bot/core/annotations/BotConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package tech.chorume.bot.core.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Essa anotação determina a classe de configuração do Bot.
+ * Ela será responsável por criar a instância do framework JDA e configurá-la.
```
Comment (brunofunnie) [inline src/main/java/tech/chorume/bot/core/annotations/BotConfiguration.java:10]: Wrong language, should use standard american english

File: src/main/java/tech/chorume/bot/core/annotations/CommandBuilder.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package tech.chorume.bot.core.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Essa anotação determina que esse Componente é um Builder de Comando e, por conta disso,
+ * será carregado pela aplicação e usado para definir os comandos do bot.
```
Comment (brunofunnie) [inline src/main/java/tech/chorume/bot/core/annotations/CommandBuilder.java:10]: Wrong language, should use standard american english

File: src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import java.util.Map;
+
+public interface DiscordBotConfiguration {
+    /**
+     * Recupera o token do Discord BOT.
+     * @return {@link String}
+     */
+    String getDiscordToken();
+    /**
+     * Determina quais intents o bot precisa.
```
Comment (brunofunnie) [inline src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java:16]: Wrong language, should use standard american english

File: src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+import net.dv8tion.jda.api.hooks.EventListener;
+import net.dv8tion.jda.api.requests.GatewayIntent;
+
+import java.util.Collection;
+import java.util.Map;
+
+public interface DiscordBotConfiguration {
+    /**
+     * Recupera o token do Discord BOT.
```
Comment (brunofunnie) [inline src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java:11]: Wrong language, should use standard american english

File: src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+     */
+    String getDiscordToken();
+    /**
+     * Determina quais intents o bot precisa.
+     * @return {@link Collection<GatewayIntent>}
+     */
+    Collection<GatewayIntent> getGatewayIntents();
+
+    /**
+     * Determina quais listeners o bot irá iniciar e oferece, como opção, os handlers definidos na aplicação.
```
Comment (brunofunnie) [inline src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java:22]: Wrong language, should use standard american english

**Review by brunofunnie — CHANGES_REQUESTED** [review]
Wrong language, should use standard american english

File: src/main/java/tech/chorume/bot/core/DiscordBotConfigurationDefault.java (hunk 2 above)
Comment (bloiseleo) [reply src/main/java/tech/chorume/bot/core/DiscordBotConfigurationDefault.java]: Applied

File: src/main/java/tech/chorume/bot/core/annotations/BotConfiguration.java (hunk 3 above)
Comment (bloiseleo) [reply src/main/java/tech/chorume/bot/core/annotations/BotConfiguration.java]: Applied

File: src/main/java/tech/chorume/bot/core/annotations/CommandBuilder.java (hunk 4 above)
Comment (bloiseleo) [reply src/main/java/tech/chorume/bot/core/annotations/CommandBuilder.java]: Applied

File: src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java (hunk 5 above)
Comment (bloiseleo) [reply src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java]: Applied

File: src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java (hunk 6 above)
Comment (bloiseleo) [reply src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java]: Applied

File: src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java (hunk 7 above)
Comment (bloiseleo) [reply src/main/java/tech/chorume/bot/core/interfaces/DiscordBotConfiguration.java]: Applied

**Review by brunofunnie — APPROVED** [review]

**Pull request merged**